pkg/prometheus/server/statefulset.go

```diff
@@ -535,6 +535,7 @@ func makeStatefulSetSpec(
 
 // appendServerArgs appends arguments that are only valid for the Prometheus server.
 func appendServerArgs(
+	cpf monitoringv1.CommonPrometheusFields,
```

Comment:
(nit) to keep it consistent with the current signature, I would add a `walCompression *bool` parameter rather than passing a CommonPrometheusFields parameter.